### PR TITLE
Fixes #1774

### LINKED ISSUE #1774: Reply button with usernames containing quotes
When using the reply button to quote a post, the quote MyCode inserted in the editor is wrong if the quoted user has an apostrophe in their username.

**Example:** Username: "Fardel's Bear"
**Action:** Click Reply on a post by the user
**Created quote code in the full editor:**

```
[quote=Fardel pid='1' dateline='1421574511']
This is a test post for testing MyAlerts...
[/quote]
```

I've not looked at the code, but I can confirm this on a  fresh install of the current dev version of MyBB 1.8 and a live forum running 1.8.3.

Original thread: [Reply button with usernames containing quotes](http://community.mybb.com/thread-165780.html)

### PR DESCRIPTION
**Pull request by WildcardSearch**
Fixes #1774 where single quotes were used to quote the user name without checking
to see if the user name contains single quotes. If it does, this causes
the quote MyCode to fail.

### CODE REVIEW

**Comment by WildcardSearch** [comment]
@mybb/sqa can someone take a look at this? Thanks.

**Comment by Shade-** [comment]
Tested. Working fine.

File: inc/functions_posting.php [hunk 1]
```diff
@@ -214,6 +214,11 @@ function parse_quoted_message(&$quoted_post, $remove_message_quotes=true)
 		$extra = " pid='{$quoted_post['pid']}' dateline='{$quoted_post['dateline']}'";
 	}
 
-	return "[quote='{$quoted_post['username']}'{$extra}]\n{$quoted_post['message']}\n[/quote]\n\n";
+	$quote_char = '"';
+	if(strpos($quoted_post['username'], '"') !== false)
+	{
+		$quote_char = "'";
+	}
```
Comment (Shade-) [inline inc/functions_posting.php]: Just one styling thing, wouldn't be better to add a newline after this bracket?

Comment (WildcardSearch) [inline inc/functions_posting.php]: No problem. I'll add it after work today.

**Comment by WildcardSearch** [comment]
@Shade- that should have this ready to merge.

**Pull request merged**